cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/mds/DatasetInstanceMDS.java

```diff
@@ -67,16 +68,34 @@ public boolean delete(DatasetId datasetInstanceId) {
   }
 
   public Collection<DatasetSpecification> getAll(NamespaceId namespaceId) {
-    Predicate<DatasetSpecification> localDatasetFilter = new Predicate<DatasetSpecification>() {
+    return getAll(namespaceId, new HashMap<String, String>());
+  }
+
+  public Collection<DatasetSpecification> getAll(NamespaceId namespaceId, final Map<String, String> properties) {
+    Predicate<DatasetSpecification> filter = new Predicate<DatasetSpecification>() {
       @Override
       public boolean apply(@Nullable DatasetSpecification input) {
-        return input != null
-          && !Boolean.parseBoolean(input.getProperty(Constants.AppFabric.WORKFLOW_LOCAL_DATASET_PROPERTY));
+        if (input == null) {
+          return false;
+        }
+        if (properties.isEmpty()) {
+          // Filter local datasets when no explicit property based filter is specified
```

Comment:
This doesn't seem right. Should have a method that takes Predicate instead. This filtering logic should only exists in the old `getAll` method, not in a more generic method.

Reply:
fixed.